### ISSUE TITLE
EKS facilitator jobs incurring 30 second timeout every time they authenticate to S3?

### ISSUE THREAD

**Issue by tgeoghegan** [abetterinternet/prio-server#256]
Derrick from NCI shared an interesting log from a successful run of one of their batch intake jobs.

```
kubectl -n ta-ta logs i-batch-926ab7a0-8fdc-31a2-a837-309457f00dc2-wlg48
[2020-12-02T00:36:35Z INFO  facilitator] 0.1.0 UNKNOWN 2020-11-18T21:55:08.869663145+00:00
[2020-12-02T00:36:36Z INFO  ureq::unit] sending request GET https://www.gstatic.com/prio-manifests/global-manifest.json
[2020-12-02T00:36:37Z INFO  ureq::unit] sending request GET https://gamlin-test.manifests.isrg-prio.org/ta-ta-g-enpa-manifest.json
[2020-12-02T00:36:37Z INFO  facilitator::intake] processing intake from s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-ingestion/ and saving validity to s3://us-east-1/prio-gamlin-test-ta-ta-g-enpa-peer-validation-a/ and s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-own-validation/
[2020-12-02T00:36:37Z INFO  facilitator::transport::s3] get s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-ingestion//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.batch.sig as None
[2020-12-02T00:37:07Z INFO  facilitator::transport::s3] get s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-ingestion//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.batch as None
[2020-12-02T00:37:38Z INFO  facilitator::transport::s3] get s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-ingestion//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.batch.avro as None
[2020-12-02T00:38:08Z INFO  facilitator::transport::s3] put s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-own-validation//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.validity_0.avro as None
[2020-12-02T00:38:39Z INFO  facilitator::transport::s3] put s3://us-east-1/prio-gamlin-test-ta-ta-g-enpa-peer-validation-a//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.validity_0.avro as None
[2020-12-02T00:39:15Z INFO  facilitator::transport::s3] put s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-own-validation//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.validity_0 as None
[2020-12-02T00:39:45Z INFO  facilitator::transport::s3] put s3://us-east-1/prio-gamlin-test-ta-ta-g-enpa-peer-validation-a//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.validity_0 as None
[2020-12-02T00:40:15Z INFO  facilitator::transport::s3] put s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-own-validation//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.validity_0.sig as None
[2020-12-02T00:40:46Z INFO  facilitator::transport::s3] put s3://us-east-1/prio-gamlin-test-ta-ta-g-enpa-peer-validation-a//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.validity_0.sig as None
```

We don't get a ton of info, but we can see that each request to S3 takes about 30 seconds, regardless of whether it's a GET or PUT, or how large the content is. The first one, for `s3://us-east-1/pha-narnia-ta-ta-ta-g-enpa-ingestion//PeriodicExposureNotification-v1/2020/12/01/23/30/926ab7a0-8fdc-31a2-a837-309457f00dc2.batch.sig`, is just a 400 byte file, so it makes no sense that it should take so long. The fact that it's so consistently 30 seconds makes me think we're consistently waiting for something to time out before falling back to something that works. [This line](https://github.com/abetterinternet/prio-server/blob/3688470a3b1b9d390f2671b2c08dba8b46df7a3b/facilitator/src/transport/s3.rs#L820) in s3.rs makes me very suspicious.

**Comment by jsha** [comment]
Maybe try turning RUST_LOG up to `trace`? It looks like reqwest does have some logging but tucks most of it away at the trace level.

**Comment by tgeoghegan** [comment]
Fixed in #269
